subsys/usb/class/netusb/function_ecm.c

```diff
@@ -197,10 +198,60 @@ static struct usb_ep_cfg_data ecm_ep_data[] = {
 	},
 };
 
+static inline void ecm_status_interface(u8_t *iface)
+{
+	SYS_LOG_DBG("iface %u", *iface);
+
+	/* First interface is CDC Comm interface */
+	if (*iface != NETUSB_IFACE_IDX + 1) {
+		return;
+	}
+
+	netusb_enable();
+}
+
+static void ecm_status_cb(enum usb_dc_status_code status, u8_t *param)
+{
+	/* Check the USB status and do needed action if required */
+	switch (status) {
+	case USB_DC_ERROR:
+		SYS_LOG_DBG("USB device error");
+		break;
+	case USB_DC_RESET:
+		SYS_LOG_DBG("USB device reset detected");
+		break;
+	case USB_DC_CONNECTED:
+		SYS_LOG_DBG("USB device connected");
+		break;
+	case USB_DC_CONFIGURED:
+		SYS_LOG_DBG("USB device configured");
+		break;
+	case USB_DC_DISCONNECTED:
+		SYS_LOG_DBG("USB device disconnected");
```

Comment:
@finikorg What do you think about removing unused cases, so that e.g. only USB_DC_DISCONNECTED and USB_DC_INTERFACE remain?